code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -238,24 +238,31 @@
 
 
 /obj/item/device/soulstone/proc/getCultGhost(obj/item/device/soulstone/C, mob/living/carbon/human/T, mob/U)
+	var/success = 0
+	var/client/chosen_ghost
 
-	var/list/consenting_candidates = pollCandidates("Would you like to play as a Shade?", be_special_flag = ROLE_CULTIST, poll_time = 100)
+	for(var/mob/dead/observer/ghost in player_list) //We put them back in their body
+		if(T.real_name == ghost.real_name && ghost.client)
```

Comment:
Using names could cause issues if two mobs have the same name, I think. Maybe use `if(T == ghost.mind.current` like the reenter_corpse() proc does.